documentation/src/main/asciidoc/reference/introduction.adoc

```diff
@@ -0,0 +1,517 @@
+with a reactive `Session` most of the time. Just to make things a little more
+confusing for new users, the reactive `Session` and its related interfaces all
+come in two flavors:
+
+- `Stage.Session` and friends provide a reactive API based around Java's
+`CompletionStage`, and
+- `Mutiny.Session` and friends provide an API based on {Mutiny}[Mutiny].
+
+You'll need to decide which API you want to use!
+
+TIP: If you take the time to look over the types `Stage.Session` and
+`Mutiny.Session`, you'll notice they're almost identical. Choosing between
+them is a matter of deciding which reactive API you want to use for working
+with reactive streams. Your decision won't affect what you can do with
+Hibernate Reactive.
+
+=== Obtaining a reactive session factory
+
+Whatever you decide, the first step to getting a reactive session is to obtain
+a JPA `EntityManagerFactory` just as you usually would in plain ol' regular JPA,
+for example, by calling:
+
+[java]
+----
+EntityManagerFactory emf = Persistence.createEntityManagerFactory("example");
+----
+
+Now, `unwrap()` the reactive `SessionFactory`. If you want to use
+``CompletionStage``s for chaining reactive operations, ask for a
+`Stage.SessionFactory`:
+
+[java]
+----
+Stage.SessionFactory sessionFactory = emf.unwrap(Stage.SessionFactory.class);
+----
+
+Or, if you prefer to use the Mutiny-based API, `unwrap()` the type
+`Mutiny.SessionFactory`:
+
+[java]
+----
+Mutiny.SessionFactory sessionFactory = emf.unwrap(Mutiny.SessionFactory.class);
+----
+
+Reactive sessions may be obtained from the resulting reactive `SessionFactory`.
+
+TIP: It's also possible to construct a reactive `SessionFactory` via programmatic
+configuration based on Hibernate's `ServiceRegistry` architecture, by using a
+`ReactiveServiceRegistryBuilder`. But that's outside the scope of this document.
+
+=== Obtaining a reactive session
+
+Persistence operations are exposed via a reactive `Session` object. It's very
+important to understand that most operations of this interface are non-blocking,
+and execution of SQL against the database is never performed synchronously.
+Persistence operations that belong to a single unit of work must be chained by
+composition within a single reactive stream.
+
+Also remember that a Hibernate session is a lightweight object that should be
+created, used, and then discarded within a single logical unit of work.
+
+TIP: That is to say, you should reuse the same session across multiple
+persistence operations within a single reactive stream representing a certain
+unit of work, but don't share a session between different reactive streams!
+
+To obtain a reactive `Session` from the `SessionFactory`, use `withSession()`:
+
+[java]
+----
+sessionFactory.withSession(
+        session -> ... //do some work
+);
+----
+
+Alternatively, you may use `openSession()`, but you must remember to `close()`
+the session when you're done.
```

Comment:
I think it's worth mentioning that `flush()` operations aren't done by default. Should we mention `withTransaction()`?

Reply:
I do so in the very next section.